hibernate-core/src/main/java/org/hibernate/bytecode/enhance/internal/bytebuddy/PersistentAttributeTransformer.java

```diff
@@ -187,15 +192,19 @@ private boolean isEnhanced(String owner, String name, String desc) {
 							enhancedField.getType().asErasure(),
 							Visibility.PUBLIC
 					)
-					.intercept( fieldReader( enhancedField )
-					)
-					.defineMethod(
-							EnhancerConstants.PERSISTENT_FIELD_WRITER_PREFIX + enhancedField.getName(),
-							TypeDescription.VOID,
-							Visibility.PUBLIC
-					)
-					.withParameters( enhancedField.getType().asErasure() )
-					.intercept( fieldWriter( enhancedField ) );
+					.intercept( fieldReader( enhancedField ) );
+			// Final fields will only be written to from the constructor,
+			// so there's no point trying to replace final field writes with a method call.
+			if ( !enhancedField.getFieldDescription().isFinal() ) {
+				builder = builder
+						.defineMethod(
```

Comment:
Do you rememer if Panache is taking the existance of such methods for granted?

Reply:
It is. However, I think Panache only performs extended enhancement, and I think this enhancement supposedly avoids transforming field access within the class declaring the field?

Not sure if it's trying or succeeding, though; ORM is trying but failing, as far as I can see, but that's another problem.

In any case, we'll have to add tests in Quarkus. I'll submit a PR.